src/Microsoft.Azure.WebJobs.Host/Tables/TableAttributeBindingProvider.cs

```diff
@@ -365,5 +425,76 @@ public ResolvedTableAttribute(TableAttribute inner, IStorageTableClient client)
 
             internal IStorageTableClient Client { get; private set; }
         }
+
+        // IStorageTable --> IQueryable<T>
+        // ConverterManager's pattern matcher will figure out TElement. 
+        private class Table2IQueryableConverter<TElement> where TElement : ITableEntity, new()
+        {
+            public Table2IQueryableConverter()
+            {
+                // We're now commited to an IQueryable. Verify other constraints. 
+                Type entityType = typeof(TElement);
+
+                if (!TableClient.ImplementsITableEntity(entityType))
+                {
+                    throw new InvalidOperationException("IQueryable is only supported on types that implement ITableEntity.");
+                }
+
+                TableClient.VerifyDefaultConstructor(entityType);
+            }
+
+            public IQueryable<TElement> Convert(IStorageTable value)
+            {
+                // If Table does not exist, treat it like have zero rows. 
+                // This means return an non-null but empty enumerable.
+                // SDK doesn't do that, so we need to explicitly check. 
+                Task<bool> t = Task.Run(() => value.ExistsAsync(CancellationToken.None));
+                bool exists = t.GetAwaiter().GetResult();
+
+                if (!exists)
+                {
+                    return Enumerable.Empty<TElement>().AsQueryable();
+                }
+                else
+                {
+                    return value.CreateQuery<TElement>();
+                }
+            }
+        }
+
+        // Convert from T --> ITableEntity
+        private class Object2ITableEntityConverter<TElement>
+        {
+            private static readonly IConverter<TElement, ITableEntity> Converter = PocoToTableEntityConverter<TElement>.Create();
+
+            public Object2ITableEntityConverter()
+            {
+                // JObject case should have been claimed by another converter. 
+                // So we can statically enforce an ITableEntity compatible contract
+                var t = typeof(TElement);
+                TableClient.VerifyContainsProperty(t, "RowKey");
+                TableClient.VerifyContainsProperty(t, "PartitionKey");
+            }
+
+            public ITableEntity Convert(TElement item)
+            {
+                return Converter.Convert(item);
+            }
+        }
+
+        // Matches IQueryable<T> where T is derived from ITableEntity
+        private class TableQueryableOpenType : OpenType
+        {
+            public static bool IsValid(Type t)
```

Comment:
The empty OpenType marker type still feels weird to me. If IsValid is required, why isn't is part of the base class?